File: JULY-2025/27-07-2025/p2210.swift
class Solution {
    func countHillValley(_ nums: [Int]) -> Int {
        nums
            .chunked(by: ==)
            .compactMap(\.first)
            .windows(ofCount: 3)
            .map(Array.init)
            .count { $0[1] < min($0[0], $0[2]) || $0[1] > max($0[0], $0[2]) }
    }
}

File: JULY-2025/27-07-2025/p2210.cpp
class Solution {
public:
    int countHillValley(vector<int>& nums)
    {
        int count = 0;
        int i = 1;
        while(i < (nums.size()-1))
        {
            int first = i;
            while(i+1 < (nums.size()-1) && nums[i+1] == nums[first]) i++;
            if(i == (nums.size()-1)) break;
            int last = i;
            if((nums[first] > nums[first-1] && nums[last] > nums[last+1]) || (nums[first] < nums[first-1] && nums[last] < nums[last+1])) count++;
            i++;
        }
        return count;
    }
};